src/components/Table/index.tsx

```diff
@@ -0,0 +1,302 @@
+		) => {
+			if (!sort) {
+				return;
+			}
+			const isAsc = sort.key === column.key && sort.direction === 'asc';
+			const newOrder = isAsc ? 'desc' : 'asc';
+			// Passing the entire column is not possible because the label might be an HTML element.
+			// This can cause errors when attempting to save it to localStorage.
+			const sortObj: TableSortOptions = {
+				direction: newOrder,
+				field: column.field,
+				key: column.key,
+				refScheme: column.refScheme,
+			};
+			onSort?.(sortObj);
+		},
+		[onSort, sort],
+	);
+
+	const handleToggleCheck = React.useCallback(
+		(row: T) => {
+			return (
+				event: React.MouseEvent<HTMLButtonElement, MouseEvent> & {
+					target: { checked: boolean };
+				},
+			) => {
+				if (!visibleRowsMap) {
+					return;
+				}
+				const lastSelectedRow = lastSelected.current;
+				const isDifferentRowSelected =
+					lastSelectedRow && lastSelectedRow[rowKey] !== row[rowKey];
+				const isShiftClick =
+					event.shiftKey &&
+					isDifferentRowSelected &&
+					visibleRowsMap.has(lastSelectedRow[rowKey]);
+				// handle multiple selection
+				if (isShiftClick) {
+					let isInSelection = false;
+					for (const [key, value] of visibleRowsMap) {
+						if (row[rowKey] === key || lastSelectedRow[rowKey] === key) {
+							isInSelection = !isInSelection;
+						}
+						if (
+							(isInSelection || row[rowKey] === key) &&
+							event.target.checked
+						) {
+							checkedRowsMap.set(key, value);
+						} else if (!event.target.checked) {
+							checkedRowsMap.delete(key);
+						}
+					}
+					lastSelected.current = undefined;
+				} else {
+					// handle single selection
+					if (event.target.checked) {
+						checkedRowsMap.set(row[rowKey], row);
+					} else {
+						checkedRowsMap.delete(row[rowKey]);
+					}
+					lastSelected.current = row;
+				}
+
+				const filteredArray = Array.from(checkedRowsMap.values());
+				onCheck?.(filteredArray, filteredArray.length > 0 ? 'some' : 'none');
+			};
+		},
+		[visibleRowsMap, checkedRowsMap, rowKey, onCheck],
+	);
+
+	return (
+		<Box sx={{ width: '100%' }}>
+			<Paper sx={{ width: '100%', mb: 2 }}>
+				<TableToolbar
+					numSelected={numSelected}
+					columns={columns}
+					actions={actions}
+					onColumnPreferencesChange={onColumnPreferencesChange}
+				/>
+				<TableContainer sx={{ maxHeight: '60vh' }}>
+					<StyledMaterialTable stickyHeader>
+						<TableHead
+							data-display="table-head"
+							columns={columns}
+							numSelected={numSelected}
+							checkedState={checkedState}
+							order={sort.direction}
+							orderBy={sort.key}
+							onSelectAllClick={onCheck}
+							onRequestSort={handleOnSort}
+							rowCount={totalItems}
+						/>
+						<TableBody data-display="table-body">
+							{visibleRows.map((row, index) => {
+								const labelId = `enhanced-table-checkbox-${index}`;
+								const checked = isChecked(row);
+
+								return (
+									<TableRow
+										hover
```

Comment:
This seems odd, why do we set it as always true?

Reply:
Ah I guess `hover` makes it float over the rest of the table cells? I thought it was for an effect on-hover